plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,506 @@
+package logstash
+
+import (
+	"encoding/json"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"net/http"
+	"net/url"
+	"time"
+
+	jsonParser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+// ##### Interface #####
+
+const sampleConfig = `
+  ## This plugin reads metrics exposed by Logstash Monitoring API.
+  ## https://www.elastic.co/guide/en/logstash/current/monitoring.html
+
+  ## The URL of the exposed Logstash API endpoint
+  url = "http://127.0.0.1:9600"
+
+  ## Enable Logstash 6+ multi-pipeline statistics support
+  multi_pipeline = true
+
+  ## Should the general process statistics be gathered
+  collect_process_stats = true
+
+  ## Should the JVM specific statistics be gathered
+  collect_jvm_stats = true
+
+  ## Should the event pipelines statistics be gathered
+  collect_pipelines_stats = true
+
+  ## Should the plugin statistics be gathered
+  collect_plugins_stats = true
+
+  ## Should the queue statistics be gathered
+  collect_queue_stats = true
+
+  ## HTTP method
+  # method = "GET"
+
+  ## Optional HTTP headers
+  # headers = {"X-Special-Header" = "Special-Value"}
+
+  ## Override HTTP "Host" header
+  # host_header = "logstash.example.com"
```

Comment:
Even though the Host header must be set in a special way, let's allow the user to set it as a normal header.  This will be similar to the http output:
```
	for k, v := range h.Headers {
		if strings.ToLower(k) == "host" {
			req.Host = v
			continue
		}
		req.Header.Set(k, v)
	}
```